File: libnetwork/src/main/java/com/apache/fastandroid/network/interceptor/ErrorInterceptor.java
package com.apache.fastandroid.network.interceptor;

import org.jetbrains.annotations.NotNull;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by Jerry on 2021/3/23.
 */
public class ErrorInterceptor implements Interceptor {
    public static final String TAG = ErrorInterceptor.class.getSimpleName();
    @NotNull
    @Override
    public Response intercept(@NotNull Chain chain) throws IOException {
        Request request = chain.request();
        String url = request.url().toString();

        Request originRequest = chain.request();
        Request.Builder builder = chain.request().newBuilder();

        if (url.contains("article/top/json")){
//            throw new IllegalArgumentException("invalid url" + originRequest.url());
        }
        return chain.proceed(builder.build());
    }
}

File: libnetwork/src/main/java/com/apache/fastandroid/network/interceptor/ErrorInterceptor.kt
package com.apache.fastandroid.network.interceptor

import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

/**
 * Created by Jerry on 2021/3/23.
 */
class ErrorInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
        val response = chain.proceed(builder.build())
        when (response.code) {
            400 -> {
                //Show Bad Request Error Message
            }
            401 -> {
                //Show UnauthorizedError Message
            }
            403 -> {
                //Show Forbidden Message
            }
            404 -> {
                //Show NotFound Message
            }
            // ... and so on
        }

        return response
    }

}